src/main/java/com/github/javachat/doublerange/DoubleRange.java

```diff
@@ -0,0 +1,240 @@
+            throw new IllegalArgumentException(BOUNDARY_IS_NAN);
+        }
+
+        if (Double.compare(lowerBound, upperBound) > 0) {
+            throw new IllegalArgumentException(String.format(ILLEGAL_BOUND, lowerBound, upperBound));
+        }
+
+        if (lowerBound == upperBound
+                && lowerBoundType == BoundType.OPEN
+                && lowerBoundType == upperBoundType) {
+            throw new IllegalArgumentException(ILLEGAL_OPEN_RANGE);
+        }
+
+        this.lowerBound = lowerBound;
+        this.upperBound = upperBound;
+        this.lowerBoundType = lowerBoundType;
+        this.upperBoundType = upperBoundType;
+
+        if (Double.isInfinite(lowerBound)) {
+            this.lowerCheck = value -> true;
+        } else {
+            this.lowerCheck = (lowerBoundType == BoundType.OPEN)
+                    ? value -> value >= lowerBound
+                    : value -> value > lowerBound;
+        }
+
+        if (Double.isInfinite(upperBound)) {
+            this.upperCheck = value -> true;
+        } else {
+            this.upperCheck = (upperBoundType == BoundType.OPEN)
+                    ? value -> value <= upperBound
+                    : value -> value < upperBound;
+        }
+    }
+
+    private DoubleRange(DoubleCut lower, DoubleCut upper) {
+        this(lower.getValue(), lower.getBoundType(), upper.getValue(), upper.getBoundType());
+    }
+
+    /**
+     * Returns true if the range has a lower bound which is not negative infinity.
+     */
+    public boolean hasLowerBound() {
+        return Double.isFinite(lowerBound);
+    }
+
+    /**
+     * Returns true if the range has an upper bound which is not positive infinity.
+     */
+    public boolean hasUpperBound() {
+        return Double.isFinite(upperBound);
+    }
+
+    public BoundType lowerBoundType() {
+        return lowerBoundType;
+    }
+
+    public BoundType upperBoundType() {
+        return upperBoundType;
+    }
+
+    // TODO: Keep using our nomenclature or Guava's?
+    public double lowerBound() {
+        return lowerBound;
+    }
+
+    public double upperBound() {
+        return upperBound;
+    }
+
+    public boolean isEmpty() {
+        return lowerBound == upperBound;
+    }
+
+    private DoubleCut toLowerCut() {
+        return new DoubleCut(lowerBound, lowerBoundType);
+    }
+
+    private DoubleCut toUpperCut() {
+        return new DoubleCut(upperBound, upperBoundType);
+    }
+
+    /**
+     * Returns true if the given value is strictly contained within the range.
+     */
+    public boolean contains(double value) {
+        return lowerCheck.test(value) && upperCheck.test(value);
+    }
+
+    public boolean containsAll(double... values) {
+        return Arrays.stream(values)
+                .allMatch(this::contains);
+    }
+
+    /**
+     * Returns true if the bounds of <code>other</code> do not extend outside of the bounds of this range.
+     * <p>
+     * It follows the semantics laid out in it's
```

Comment:
The custom JavaDoc has been replaced in the mean time and replaced with links to Guava's JavaDoc